services/app-web/src/hooks/useContractForm.ts

```diff
@@ -0,0 +1,258 @@
+                return new Error('Failed to create form data')
+            }
+
+            return createdSubmission
+        } catch (serverError) {
+            setShowPageErrorMessage(true)
+            recordJSException(
+                `StateSubmissionForm: Apollo error reported. Error message: ${serverError.message}`
+            )
+            return new Error(serverError)
+        }
+    }
+    const [updateFormData] = useUpdateContractDraftRevisionMutation()
+
+    const updateDraft: UseContractForm['updateDraft']  = async (
+        input: UpdateContractDraftRevisionInput
+    ): Promise<Contract | Error> => {
+
+        setShowPageErrorMessage(false)
+        if (input.formData.contractDocuments && input.formData.contractDocuments.length > 0) {
+            input.formData.contractDocuments = documentsInput(input.formData.contractDocuments)
+        } if (input.formData.supportingDocuments && input.formData.supportingDocuments.length > 0) {
+            input.formData.supportingDocuments = documentsInput(input.formData.supportingDocuments)
+        } if (input.formData.stateContacts && input.formData.stateContacts.length > 0) {
+            input.formData.stateContacts = stateContactsInput(input.formData.stateContacts)
+        }
+
+        try {
+            const updateResult = await updateFormData({
+                variables: {
+                    input: {
+                        contractID: contractID ?? 'new-draft',
+                        lastSeenUpdatedAt: contract!.draftRevision!.updatedAt,
+                        formData: input.formData
+                    },
+                },
+            })
+            const updatedSubmission =
+                updateResult?.data?.updateContractDraftRevision.contract
+            if (!updatedSubmission) {
+                setShowPageErrorMessage(true)
+                recordJSException(
+                    `StateSubmissionForm: Apollo error reported. Error message: Failed to update form data ${updateResult}`
+                )
+                return new Error('Failed to update form data')
+            }
+            return updatedSubmission
+        } catch (serverError) {
+            setShowPageErrorMessage(true)
+            recordJSException(
+                `StateSubmissionForm: Apollo error reported. Error message: ${serverError.message}`
+            )
+            return new Error(serverError)
+        }
+    }
+    const { 
+        data: fetchResultData,
+        error: fetchResultError,
+        loading: fetchResultLoading
+     } = useFetchContractQuery({
+        variables: {
+            input: {
+                contractID: contractID ?? 'new-draft'
+            }
+        },
+        skip: !contractID
+    })
+    const contract = fetchResultData?.fetchContract.contract
+    if (fetchResultLoading) {
+        interimState = 'LOADING'
+        return {interimState, createDraft, updateDraft, showPageErrorMessage }
+    }
+    if (fetchResultError) {
```

Comment:
I found I had to recreate these fields explicitly because these particular fields that came back from `fetchContract` were set to being possibly undefined and unlockedType doesn't accept undefined for them

Reply:
I'll look into using the apolloQueryWrapper. I think useHealthPlanForm uses it iirc